File: inst/htmlwidgets/g2percent_stacked_area.js
HTMLWidgets.widget({
  name: "g2percent_stacked_area",
  type: "output",
  factory: function(el, width, height) {
    var chart;
    return {
      renderValue: function(x) {
        chart = new G2Plot.PercentStackedArea(el.id, x.cfg);
        chart.render();
      }
    };
  }
});

File: R/g2PercentStackedArea.R
#' PercentStackedArea chart
#' 
#' PercentStackedArea chart
#' 
#' @param xField,yField,colorField column name in data for aesthetic mapping
#' @inheritParams g2
#' @family bar
#' 
#' @export
g2PercentStackedArea <- function(data, xField, yField, colorField = '', cfg = list(), width = NULL, height = NULL) {
  # prep cfg
  cfg$data       = jsonlite::toJSON(data)
  cfg$xField     = xField
  cfg$yField     = yField
  cfg$colorField = colorField
  
  # pass the data and settings using 'x'
  x <- list(
    type = 'PercentStackedArea',
    cfg = cfg
  )
  # create the widget
  htmlwidgets::createWidget('g2', x, width = width, height = height, package='rg2')
}



